src/test/java/com/slisenko/education/xml/XMLParserTest.java

```diff
@@ -49,165 +49,198 @@ public void parseStr() {
+                "<students>" +
+                    "<student>" +
+                        "<name>Name3</name><surname>Surname3</surname><year>1993</year><faculty>Faculty3</faculty><course>3</course>" +
+                    "</student>" +
+                    "<student>" +
+                        "<name>Name2</name><surname>Surname2</surname><year>1992</year><faculty>Faculty2</faculty><course>2</course>" +
+                    "<????>" +
+                    "<student>" +
+                        "<name>Name1</name><surname>Surname1</surname><year>1991</year><faculty>Faculty1</faculty><course>1</course>" +
+                    "</student>" +
+                "</students>";
 
-        String badXml = strXml.substring(0,  index) + "</????>" +
-                strXml.substring(index + tag.length());
-        /*
-        String badXml = strXml.substring(0,  index) +
-                strXml.substring(index + tag.length());
-        */
         List<Student> expected = new ArrayList<>();
-
         Student student = new Student();
 
-        student.setName("Имя 3");
-        student.setSurname("Фамилия 3");
+        student.setName("Name3");
+        student.setSurname("Surname3");
         student.setYear(1993);
-        student.setFaculty("Факультет 3");
+        student.setFaculty("Faculty3");
         student.setCourse(3);
         expected.add(student);
 
         XMLParser parser = new XMLParser();
         List<Student> actual = parser.parseStr(badXml);
-        //List<Student> actual = parser.parseStr(strXml);
 
         Assert.assertEquals(expected, actual);
     }
 
     @Test
     public void parseStr_BadTableOpenTag() {
 
-        String tag = "<students>";
-        int index = strXml.indexOf(tag);
-        String badXml = strXml.substring(0,  index) + "<????>" +
-                strXml.substring(index + tag.length());
-        /*
-        String badXml = strXml.substring(0,  index) +
-                strXml.substring(index + tag.length());
-        */
+        // Bad tag <students> -> <????>
+        String badXml =
+                "<????>" +
+                    "<student>" +
+                        "<name>Name3</name><surname>Surname3</surname><year>1993</year><faculty>Faculty3</faculty><course>3</course>" +
+                    "</student>" +
+                    "<student>" +
+                        "<name>Name2</name><surname>Surname2</surname><year>1992</year><faculty>Faculty2</faculty><course>2</course>" +
+                    "</student>" +
+                    "<student>" +
+                        "<name>Name1</name><surname>Surname1</surname><year>1991</year><faculty>Faculty1</faculty><course>1</course>" +
+                    "</student>" +
+                "</students>";
+
         XMLParser parser = new XMLParser();
         List<Student> list = parser.parseStr(badXml);
-        //List<Student> list = parser.parseStr(strXml);
 
-        Assert.assertTrue(list.isEmpty());
+        if (!list.isEmpty()) {
```

Comment:
use assertFalse, fix in other places like this

Reply:
Why assertFalse? I think necessary assertTrue(list.isEmpty()).

Reply:
ok, that works